### DISCUSSION TITLE
Custom Joi Validation for JWT Tokens

### DISCUSSION THREAD

**Discussion by huzaifa-99** [hagopj13/node-express-boilerplate#115] [General]
Hi @hagopj13. I have been using the boilerplate for some time and have been really enjoying it. 

Regarding the Joi Validation. The boilerplate requires the user to send JWT tokens either with req.body or req.query for some routes like `auth/reset-password` or `auth/logout`, but they are not validated to have a basic JWT structure. For example:  `token=abc` will pass the validation. 

In contrast, we do this type of validation for a valid mongo id in [custom.validation.js](https://github.com/hagopj13/node-express-boilerplate/blob/master/src/validations/custom.validation.js). A custom validation method can be added in [custom.validation.js](https://github.com/hagopj13/node-express-boilerplate/blob/master/src/validations/custom.validation.js) for validating JWT.

**Comment by hagopj13** [comment]
@huzaifa-99 thanks for the idea. Would you like to do a PR on it?

**Reply by huzaifa-99** [reply]
Sure
